test/resources/models/inference_example.py

```diff
@@ -0,0 +1,198 @@
+BASE_URI = 'http://localhost:8080/invocations'
+
+JSON_CONTENT_TYPE = ['application/json', 'application/jsons', 'application/jsonlines']
+
+
+def input_handler(data, context):
+    """ Pre-process request input before it is sent to TensorFlow Serving REST API
+
+    Args:
+        data (obj): the request data, in format of dict or string
+        context (Context): an object containing request and configuration details
+
+    Returns:
+        (dict): a JSON-serializable dict that contains request body and headers
+    """
+
+    if context.request_content_type == 'application/json':
+        data = _parse_json(data)
+    elif context.request_content_type == 'application/jsons' or \
+            context.request_content_type == 'application/jsonlines':
+        data = _parse_jsonlines(data)
+
+        print('*********** parsed jsonlines: ' + str(data))
+    elif context.request_content_type == 'text/csv':
+        data = _parse_csv(data)
+    else:
+        _return_error(415, 'Unsupported content type "{}"'.format(context.request_content_type or 'Unknown'))
+
+    processed_input = {
+        'headers': _make_headers(context),
+        'data': data
+    }
+
+    return processed_input
+
+
+def output_handler(data, context):
+    """Post-process TensorFlow Serving output before it is returned to the client.
+
+    Args:
+        data (obj): the TensorFlow serving response
+        context (Context): an object containing request and configuration details
+
+    Returns:
+        (dict, string): data to return to client, response content type
+    """
+    processed_output = json.loads(data.text)
+    response_content_type = context.accept_header or DEFAULT_CONTENT_TYPE
+    return processed_output, response_content_type
+
+
+def handler(data, context):
+    """Pre-process request input, and post request to TensorFlow REST API, then post-process output before it is
+    sent to the client
+
+    Args:
+        data (obj): the request data, in the format of dict or string
+        context (Context): an object containing request and configuration details
+
+    Returns:
+        (dict, string): data to return to client, response content type
+    """
+    processed_input = input_handler(data, context)
+    processed_output = _post_request(processed_input, context)
+    return str(processed_output['outputs'])
+
+
+def _post_request(data, context):
+    response = requests.post(context.rest_uri, data=data['data'], headers=data['headers'])
+
+    processed_output, _ = output_handler(response, context)
+    if 'error' in processed_output:
+        raise Exception('Invocation is not successful.')
+
+    return processed_output
+
+
+def _make_headers(context):
+    headers = {
+        'Content-Type': context.request_content_type,
+        'Accept': context.accept_header,
+        'X-Amzn-SageMaker-Custom-Attributes': _parse_custom_attributes(context)
+    }
+    return headers
+
+
+def _parse_custom_attributes(context):
```

Comment:
this should be in our service code... users should have to see or implement this, they should just receive a context object that is correctly populated